model/task/db.go

```diff
@@ -2592,60 +2595,103 @@ func getTasksByVersionPipeline(versionID string, opts GetTasksByVersionOptions)
-			})
-		}
-		pipeline = append(pipeline, []bson.M{
-			// Add data about the base task
-			{"$lookup": bson.M{
-				"from": Collection,
-				"let": bson.M{
-					RevisionKey:            "$" + RevisionKey,
-					BuildVariantKey:        "$" + BuildVariantKey,
-					DisplayNameKey:         "$" + DisplayNameKey,
-					RevisionOrderNumberKey: "$" + RevisionOrderNumberKey,
+	if shouldPopulateBaseTask {
+		// First group by variant and task name to group all tasks and their base tasks together
+		pipeline = append(pipeline, bson.M{
+			"$group": bson.M{
+				"_id": bson.M{
+					BuildVariantKey: "$" + BuildVariantKey,
+					DisplayNameKey:  "$" + DisplayNameKey,
 				},
-				"as": BaseTaskKey,
-				"pipeline": []bson.M{
-					{"$match": bson.M{
-						RequesterKey: evergreen.RepotrackerVersionRequester,
-						"$expr": bson.M{
-							"$and": baseCommitMatch,
+				"tasks": bson.M{
+					"$push": "$$ROOT",
+				},
+			},
+		})
+		// Separate the root task and base task into separate arrays
+		pipeline = append(pipeline, bson.M{
+			"$addFields": bson.M{
+				"root_task": bson.M{
+					"$filter": bson.M{
+						"input": "$tasks",
+						"as":    "task",
+						"cond": bson.M{
+							"$eq": []string{"$$task." + VersionKey, versionID},
 						},
-					}},
-					{"$project": bson.M{
-						IdKey:     1,
-						StatusKey: displayStatusExpression,
-					}},
-					{"$limit": 1},
+					},
 				},
-			}},
-			{
-				"$unwind": bson.M{
-					"path":                       "$" + BaseTaskKey,
-					"preserveNullAndEmptyArrays": true,
+				"base_task": bson.M{
+					"$filter": bson.M{
+						"input": "$tasks",
+						"as":    "task",
+						"cond": bson.M{
+							"$eq": []string{"$$task." + VersionKey, opts.BaseVersionID},
+						},
+					},
 				},
 			},
-		}...,
-		)
+		})
+		// Unwind the root task and base task arrays so that each document is a single task
+		pipeline = append(pipeline, bson.M{
+			"$addFields": bson.M{
+				"root_task": bson.M{
+					"$first": "$root_task",
+				},
+				"base_task": bson.M{
+					"$first": "$base_task",
+				},
+			},
+		})
+
+		// Ensure that the root task is not nil if it is the task does not exist in the current version
+		pipeline = append(pipeline, bson.M{
+			"$match": bson.M{
+				"root_task": bson.M{
+					"$exists": true,
+				},
+			},
+		})
+
+		// Include the base task in the root task
+		pipeline = append(pipeline, bson.M{
+			"$addFields": bson.M{
+				"root_task": bson.M{
+					"base_task": "$base_task",
+				},
+			},
+		})
+
+		// Project out the base task and the tasks array since they are no longer needed
+		pipeline = append(pipeline, bson.M{
+			"$project": bson.M{
+				"base_task": 0,
+				"tasks":     0,
+			},
+		})
+
+		// Use display status from the base task as its status
+		pipeline = append(pipeline, bson.M{
+			"$addFields": bson.M{
+				bsonutil.GetDottedKeyName("root_task", BaseTaskKey, StatusKey): "$" + bsonutil.GetDottedKeyName("root_task", BaseTaskKey, DisplayStatusKey),
```

Comment:
I wonder if you can use `BaseTaskStatusKey` here... I would double check though lol 

```suggestion
				bsonutil.GetDottedKeyName("root_task", BaseTaskStatusKey): "$" + bsonutil.GetDottedKeyName("root_task", BaseTaskKey, DisplayStatusKey),
```